### PR TITLE
Remove non standard header findings and add deprecated headers findings

### PR DESCRIPTION
**Pull request by noamblitz**
### Changes

Remove findings from non standard headers as stated in https://github.com/minvws/nl-kat-coordination/issues/2731

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/2731

### Demo

_Please add some proof in the form of screenshots or screen recordings to show (off) new functionality, if there are interesting new features for end-users._

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — APPROVED** [review]
No remarks

**Comment by underdarknl** [comment]
Seeing that these headers are not longer required, but also are now suggested to *not* be used, we should probably just invert the findings stating these headers *should* not exists if they are found.
Just removing them isn't enough in my book.

**Comment by underdarknl** [comment]
Nice! looks good to me.

File: octopoes/bits/missing_headers/missing_headers.py [hunk 1]
```diff
@@ -104,3 +84,14 @@ def run(resource: HTTPResource, additional_oois: list[HTTPHeader], config: dict[
...
         yield finding
+
+    if set(header_keys) & DEPRECATED_HEADER:
+        ft = KATFindingType(id="KAT-DEPRECATED-HEADERS")
+        finding = Finding(
+            finding_type=ft.reference,
+            ooi=resource.reference,
+            description=f"Deprecated headers are used. Avoid using the following headers: "
+            f"{' '.join(DEPRECATED_HEADER & set(header_keys))}",
+        )
```
Comment (ammar92) [inline octopoes/bits/missing_headers/missing_headers.py:95]: It's probably better to reuse this statement by using a variable, e.g.
```suggestion
    deprecated_headers = set(header_keys) & DEPRECATED_HEADER
    if deprecated_headers:
        ft = KATFindingType(id="KAT-DEPRECATED-HEADERS")
        finding = Finding(
            finding_type=ft.reference,
            ooi=resource.reference,
            description=f"Deprecated headers are used. Avoid using the following headers: "
            f"{' '.join(deprecated_headers)}",
        )
```
Also, the f-string didn't make much sense to use here

**Comment by stephanie0x00** [comment]
As discussed with @noamblitz the title of the finding is renamed to "KAT-NONSTANDARD-HEADERS" for all these headers to include the headers listed below. In the future it would be nice to have these split into two categories: Non-standard findings and deprecated findings. 

> Seeing that these headers are not longer required, they are however still not-standard. Maybe we should introduce a finding for those (if don't already have one) and raise that for all of these non-standard headers. The following not standard headers would warrant this I'd think:
> 
>     * [X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For)
> 
>     * [X-Forwarded-Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host)
> 
>     * [X-Forwarded-Proto](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto)
> 
>     * [X-DNS-Prefetch-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-DNS-Prefetch-Control)
> 
>     * [X-Robots-Tag](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Robots-Tag)
> 
> 
> There's also the deprecated headers that might warrant a similar warning
> 
>     * [Pragma](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Pragma)
> 
>     * [Warning](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Warning)

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Manually added the various deprecated and non-standard hosts. They are now all gathered under the'Non-standard header' finding. Created a new ticket to split these into non-standard and deprecated headers, as that would be more ideal. #3188 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**